src/main.js

```diff
@@ -31,12 +36,11 @@ const args = parseCLIArgs();
 const userDataPath = getUserDataPath(args);
 app.setPath('userData', userDataPath);
 
-// Set temp directory based on crash-reporter-directory CLI argument
-// The crash reporter will store crashes in temp folder so we need
-// to change that location accordingly.
+// Configure static command line arguments
+const argvConfig = configureCommandlineSwitchesSync(args);
 
-// If a crash-reporter-directory is specified we setup the crash reporter
-// right from the beginning as early as possible to monitor all processes.
+// If a crash-reporter-directory is specified we store the crash reports
+// in the specified directory and don't upload them to the crash server.
 let crashReporterDirectory = args['crash-reporter-directory'];
 if (crashReporterDirectory) {
```

Comment:
This entire `if-else` block would probably read better if:
* the crash reporter ID comes from `argv.json` (as discussed below) so you avoid a promise
* the `crashReporter.start` method is only called once and the properties are given depending on the command line args instead of starting it from each if-else branch